benches/main.rs

```diff
@@ -0,0 +1,133 @@
+    let no_of_subscribers = 200;
+
+    let mut event_manager = EventManager::<Arc<dyn EventSubscriber + Send + Sync>>::new().unwrap();
+    for _ in 0..no_of_subscribers {
+        let counter_subscriber = CounterInnerMutSubscriber::new();
+        counter_subscriber.trigger_event();
+        event_manager.add_subscriber(Arc::new(counter_subscriber));
+    }
+
+    c.bench_function("process_with_inner_mut", |b| {
+        b.iter(|| {
+            let ev_count = event_manager.run().unwrap();
+            assert_eq!(ev_count, no_of_subscribers)
+        })
+    });
+}
+
+// Test the performance of event manager when it manages subscribers of different types, that are
+// wrapped in an Arc<Mutex>. Also, make use of `Events` with custom user data
+// (using CounterSubscriberWithData).
+// The performance is assessed under stress, all added subscribers have active events, and the
+// CounterSubscriberWithData subscribers have multiple active events.
+fn run_multiple_subscriber_types(c: &mut Criterion) {
+    let no_of_subscribers = 100;
+
+    let mut event_manager = EventManager::<Arc<Mutex<dyn MutEventSubscriber>>>::new()
+        .expect("Cannot create event manager.");
+
+    for i in 0..no_of_subscribers {
+        // The `CounterSubscriberWithData` expects to receive a number as a parameter that
+        // represents the number it can use as its inner Events data.
+        let mut data_subscriber = CounterSubscriberWithData::new(i * no_of_subscribers);
+        data_subscriber.trigger_all_counters();
+        event_manager.add_subscriber(Arc::new(Mutex::new(data_subscriber)));
+
+        let mut counter_subscriber = CounterSubscriber::new();
+        counter_subscriber.trigger_event();
+        event_manager.add_subscriber(Arc::new(Mutex::new(counter_subscriber)));
+    }
+
+    c.bench_function("process_dynamic_dispatch", |b| {
+        b.iter(|| {
+            let _ = event_manager.run().unwrap();
+        })
+    });
+}
+
+// Test the performance of event manager when it manages a single subscriber type.
+// Just a few of the events are active in this test scenario.
+fn run_with_few_active_events(c: &mut Criterion) {
+    let no_of_subscribers = 200;
+
+    let mut event_manager = EventManager::<CounterSubscriber>::new().unwrap();
+
+    for i in 0..no_of_subscribers {
+        let mut counter_subscriber = CounterSubscriber::new();
+        // Let's activate the events for a few subscribers (i.e. only the ones that are
+        // divisible by 23). 23 is a random number that I just happen to like.
+        if i % 23 == 0 {
+            counter_subscriber.trigger_event();
+        }
+        event_manager.add_subscriber(counter_subscriber);
+    }
+
+    c.bench_function("process_dispatch_few_events", |b| {
+        b.iter(|| {
+            let _ = event_manager.run().unwrap();
+        })
+    });
+}
+
+criterion_group! {
+    name = benches;
+    config = Criterion::default().sample_size(500);
```

Comment:
It looks like these settings cause benchmarks to run for a short time (around 5-7 each), which in turn seems to lead to a bit more variability than we'd want. I think it would make sense to tweak this a bit. For example, I've run the benchmarks a couple of times with: 
```rust 
config = Criterion::default().sample_size(200).measurement_time(std::time::Duration::from_secs(40));
```
On that machine, the overall benchmark duration went up to ~ 4 mins, but the variability decreased significantly.

Reply:
Done.